### DISCUSSION TITLE
How do I access the console to see errors?

### DISCUSSION THREAD

**Discussion by tekiegirl** [SilentVoid13/Templater#272] [Help] [answered]
Hi,
I am struggling to work out what is wrong with what I am writing for Templater as I don't know how to view the error information.
I can't even get the samples from the documentation to run.
Can someone tell me how to access the error info talked about in the pop up?
Thanks,
Jacqui

**Comment by totallyGreg** [comment]
Obsidian has it's own Developer Tools toggle under the 'View' menu.  Option-Command-I will toggle it for you. From there you should see the Console.

**Comment by welpdx** [comment, marked as answer]
If on windows, try Ctrl + Shift + I.  It is the same as chrome's console afaik. 
I find the console unhelpful in diagnosing errors. I recommend using a js editor (or even [this](https://beautifytools.com/javascript-validator.php)) first to check syntax and stuff pasting it in. 
I also like to use console.log() when debugging with the console.

**Comment by fahran** [comment]
Just discovered a gem: add the line `debugger;` in your template code, and you'll get a breakpoint you can use to debug in the console.

**Reply by AB1908** [reply]
Oh wow, never thought of that. Amazing tip, thanks! Does it take you to the exact line?

Would you mind creating a separate discussion with details? I think it's worthy of pinning at the very top.
